robot/control/Src/modules/RadioModule.cpp

```diff
@@ -17,45 +18,69 @@ RadioModule::RadioModule(BatteryVoltage *const batteryVoltage,
     secondRadioCS = 1;
 
     // todo fill out more kicker stuff
-    kickerCommand->isValid = false;
-    kickerCommand->lastUpdate = 0;
-    kickerCommand->shootMode = KickerCommand::ShootMode::KICK;
-    kickerCommand->triggerMode = KickerCommand::TriggerMode::OFF;
-    kickerCommand->kickStrength = 0;
-
-    motionCommand->isValid = false;
-    motionCommand->lastUpdate = 0;
-    motionCommand->bodyXVel = 0.0f;
-    motionCommand->bodyYVel = 0.0f;
-    motionCommand->bodyWVel = 0.0f;
-    motionCommand->dribbler = 0;
-
-    radioError->isValid = false;
-    radioError->lastUpdate = 0;
-    radioError->hasError = false;
+    auto kickerCommandLock = kickerCommand.unsafe_value();
+    kickerCommandLock->isValid = false;
+    kickerCommandLock->lastUpdate = 0;
+    kickerCommandLock->shootMode = KickerCommand::ShootMode::KICK;
+    kickerCommandLock->triggerMode = KickerCommand::TriggerMode::OFF;
+    kickerCommandLock->kickStrength = 0;
 
+    auto motionCommandLock = motionCommand.unsafe_value();
+    motionCommandLock->isValid = false;
+    motionCommandLock->lastUpdate = 0;
+    motionCommandLock->bodyXVel = 0.0f;
+    motionCommandLock->bodyYVel = 0.0f;
+    motionCommandLock->bodyWVel = 0.0f;
+    motionCommandLock->dribbler = 0;
+
+    auto radioErrorLock = radioError.unsafe_value();
+    radioErrorLock->isValid = false;
+    radioErrorLock->lastUpdate = 0;
+    radioErrorLock->hasError = false;
+}
+
+void RadioModule::start() {
+    link.init();
     printf("INFO: Radio initialized\r\n");
+    radioError.lock()->initialized = true;
 }
 
-void RadioModule::entry(void) {
-    // Just check to see if our robot id is valid
-    // That way we don't conflict with other robots on the network
-    // that are working
-    if (batteryVoltage->isValid && fpgaStatus->isValid && robotID->isValid)
-      link.send(*batteryVoltage, *fpgaStatus, *kickerInfo, *robotID);
-
-    // Try read
-    // Clear buffer of old packets such that we can get the lastest packet
-    // If you don't do this there is a significant lag of 300ms or more
-    while (link.receive(*kickerCommand, *motionCommand)) {
-      kickerCommand->isValid = true;
-      kickerCommand->lastUpdate = HAL_GetTick();
-
-      motionCommand->isValid = true;
-      motionCommand->lastUpdate = HAL_GetTick();
+void RadioModule::entry() {
+    {
+        auto batteryVoltageLock = batteryVoltage.lock();
+        auto fpgaStatusLock = fpgaStatus.lock();
+        auto robotIDLock = robotID.lock();
+        auto kickerInfoLock = kickerInfo.lock();
+
+        // Just check to see if our robot id is valid
+        // That way we don't conflict with other robots on the network
+        // that are working
+//        if (batteryVoltageLock->isValid && fpgaStatusLock->isValid && robotIDLock->isValid) {
```

Comment:
Depending on if the robotID is read before this or not, it's possible that this robot will connect to soccer as `robotID=0` for a single frame then swaps to the correct robotID. I believe soccer starts to act weird as more MAC<->RobotID combinations connect to soccer. The robots start to be only intermittently connected and you have to restart soccer to fix

Reply:
Fixed, this was only temporarily disabled.